lib/hashie/extensions/method_access.rb

```diff
@@ -35,6 +35,10 @@ def respond_to?(name, include_private = false)
       def method_missing(name, *args)
         return self[name.to_s] if key?(name.to_s)
         return self[name.to_sym] if key?(name.to_sym)
+        if name[-1] == '?'
```

Comment:
I think this is okay. AFAIK `Symbol#[]` has been available starting Ruby 1.9 and it maps to `#to_s[]`